crates/sui-json-rpc/src/axum_router.rs

```diff
@@ -203,6 +231,50 @@ async fn process_request<L: Logger>(
     response
 }
 
+pub fn monitored_reroute(
+    raw_params: Option<&RawValue>,
+    name: &str,
+    client_addr: SocketAddr,
+) -> SuiResult<(String, String)> {
+    match name {
+        "sui_executeTransactionBlock" => {
+            // add client IP arg to the params, as this is a router redirect
+            // from `execute_transaction_block`, which does require the client IP
+            let parsed_value: Value = serde_json::from_str(
+                raw_params
+                    .unwrap_or_else(|| panic!("Expected params for executeTransactionBlock"))
+                    .get(),
+            )
+            .expect("Failed to parse jsonrpsee params");
+
+            let Value::Array(mut params_vec) = parsed_value else {
+                panic!("Expected a JSON array");
+            };
+
+            params_vec.push(Value::String(client_addr.to_string()));
+            let name = String::from("sui_monitoredExecuteTransactionBlock");
+            Ok((
+                serde_json::to_string(&params_vec).expect("Failed to serialize params"),
+                name,
+            ))
+        }
+        "sui_monitoredExecuteTransactionBlock" => {
+            // Prevent an attacker calling it directly with a different
+            // client IP in order to bypass monitoring
+            Err(SuiError::InvalidRpcMethodError)
```

Comment:
Is there a way to make this API endpoint "undiscoverable" from the client @bmwill ?

Reply:
umm...only thing i can think of is doing some checking at the point where we re-write the other method

Reply:
I don't know if our openrpc spec generator is smart enough to skip certain methods. Since we've handling this already its probably not that big of a deal (because we'll remove jsonrpc at some point anyway)